### ISSUE TITLE
Non-existent `m2n` participant is not detected

### ISSUE THREAD

**Issue by davidscn** [precice/precice#1993]
**Describe your setup**

preCICE Version: 3.1.0rcXY

Add the versions/commit hashes of further adapter and/or bindings.

**Describe the problem**
During the creation of the compositional coupling scheme, we face the following assert, which is quite descriptive:

```bash
ASSERTION FAILED
Location:   const PtrParticipant precice::config::ParticipantConfiguration::getParticipant(const string&) const
File:       ~/precice/src/precice/config/ParticipantConfiguration.cpp:346
Expression: participant != _participants.end()
Rank:       0
Arguments:  fmt_error: argument not found
Stacktrace:
 0# getStacktrace[abi:cxx11]() at ~/precice/src/utils/stacktrace.cpp:11
 1# precice::config::ParticipantConfiguration::getParticipant(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const at ~/precice/src/precice/config/ParticipantConfiguration.cpp:346
 2# precice::cplscheme::CouplingSchemeConfiguration::checkSubstepExchangeWaveformDegree(precice::cplscheme::CouplingSchemeConfiguration::Config::Exchange const&) const at ~/precice/src/cplscheme/config/CouplingSchemeConfiguration.cpp:983
 3# precice::cplscheme::CouplingSchemeConfiguration::addDataToBeExchanged(precice::cplscheme::BiCouplingScheme&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const at ~/precice/src/cplscheme/config/CouplingSchemeConfiguration.cpp:1048
 4# precice::cplscheme::CouplingSchemeConfiguration::createParallelImplicitCouplingScheme(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) const at ~/precice/src/cplscheme/config/CouplingSchemeConfiguration.cpp:885
 5# precice::cplscheme::CouplingSchemeConfiguration::xmlEndTagCallback(precice::xml::ConfigurationContext const&, precice::xml::XMLTag&) at ~/precice/src/cplscheme/config/CouplingSchemeConfiguration.cpp:350
```
The underlying issue is that we have the following line in the xml file:

```xml
<receive-mesh name="TendonMeshTop" from="TendonSolver"/>
```

although the `TendonSolver` doesn't exist (see also further comments below).

**Expected behavior**
preCICE should tell, that the participant doesn't exist. Debugging xml files in compositional coupling schemes is challenging.

**Additional context**
I have not investigated, if the same issue occurs for 'usual' coupling schemes.

FYI @carme-hp

**Comment by davidscn** [comment]
I checked what happens in usual coupling schemes. In such cases, we run into a check:
`ERROR: There is no m2n communication configured between participants ...`. Indeed, changing solely the received mesh tag to the 'correct' entry, doesn't resolve the issue, only correcting the `m2n` tag resolves it 

```xml
-  <m2n:sockets acceptor="MuscleContraction" connector="TendonSolver" network="lo" />
+  <m2n:sockets acceptor="MuscleContraction" connector="BottomTendon" network="lo" />
```

**Comment by davidscn** [comment]
The complete (faulty file), for reference can be found here

https://github.com/opendihu/opendihu/blob/29278e2894385ec93b4e6a066076d5f178950b69/examples/electrophysiology/fibers/fibers_contraction/with_both_volume_and_surface_coupling/biceps/precice_config.xml

and the corresponding fixed version in this commit

https://github.com/opendihu/opendihu/commit/721e82ff6b672a79554514a730ad51caef8408e1

**Comment by fsimonis** [comment]
This has a similar cause than #1989 and requires #982 for a clean fix.
In `cplscheme` tags, we should be able to check Participants for correctness though.

The first bug is that we don't check if participants mentioned in coupling-schemes actually exist.
_The assertion is also buggy: `fmt_error: argument not found` means that the formatting of the message failed._

The second bug is in `ParticipantImpl::configureM2Ns`, where we use the other partner of the M2N unchecked.
This should be checked in `Configuration::xmlEndTagCallback`.

This config also crashes the config-visualizer :sweat_smile:

**Comment by fsimonis** [comment]
Closed via #1995
